azext_iot/operations/hub.py

```diff
@@ -2221,169 +2221,159 @@ def iot_get_module_connection_string(
-        if (
-            user_identity
-            and storage_authentication_type != AuthenticationType.identityBased.name
-        ):
-            raise ClientRequestError(
-                "Device import with user-assigned identities requires identity-based authentication [--storage-auth-type]"
-            )
-        # Track 2 CLI SDKs provide support for user-assigned identity objects
-        if (
-            ensure_iothub_sdk_min_version(IOTHUB_TRACK_2_SDK_MIN_VERSION)
-            and user_identity
-        ):
-            from azure.mgmt.iothub.models import (
-                ManagedIdentity,
-            )  # pylint: disable=no-name-in-module
-
-            import_request.identity = ManagedIdentity(user_assigned_identity=identity)
-        # if the user supplied a user-assigned identity, let them know they need a new CLI/SDK
-        elif user_identity:
-            raise CLIInternalError(
-                "Device import with user-assigned identities requires a dependency of azure-mgmt-iothub>={}".format(
-                    IOTHUB_TRACK_2_SDK_MIN_VERSION
-                )
-            )
+    service_sdk = _get_service_sdk(
+        cmd, hub_name, resource_group_name
+    )
+    export_job_properties = _create_export_import_job_properties(
+        job_type=JobType.exportDevices.value,
+        output_blob_container_uri=blob_container_uri,
+        include_keys=include_keys,
+        identity=identity
+    )
+    return service_sdk.jobs.create_import_export_job(export_job_properties)
 
-        return client.import_devices(
-            target["resourcegroup"],
-            hub_name,
-            import_devices_parameters=import_request,
+
+def iot_device_import(
+    cmd,
+    hub_name: str,
+    input_blob_container_uri: str = None,
+    input_blob_container_name: str = None,
+    input_storage_account_name: str = None,
+    output_blob_container_uri: str = None,
+    output_blob_container_name: str = None,
+    output_storage_account_name: str = None,
+    storage_authentication_type: str = None,
+    resource_group_name: str = None,
+    identity: str = None,
+):
+    if input_blob_container_uri is None:
+        input_blob_container_uri = _generate_blob_container_uri(
+            cmd, input_storage_account_name, input_blob_container_name, identity
         )
-    if storage_authentication_type:
-        raise CLIInternalError(
-            "Device import authentication-type properties require a dependency of azure-mgmt-iothub>=0.12.0"
+    if output_blob_container_uri is None:
+        output_blob_container_uri = _generate_blob_container_uri(
+            cmd, output_storage_account_name, output_blob_container_name, identity
         )
-    return client.import_devices(
-        target["resourcegroup"],
-        hub_name,
+    if storage_authentication_type is not None:
+        logger.warning(
+            "The parameter --auth-type/--storage-authentication-type has been deprecated and should not be provided"
```

Comment:
"--auth-type" being an alias for "--storage-authentication-type" is problematic because all IoT Hub CLI dataplane commands support an `--auth-type` parameter that controls how client access to the hub is achieved (key vs login/aad)...it would make sense  for `--auth-type` to work like that here when the command is using the dataplane operation.